src/plugins/content_management/server/core/event_types.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+export interface GetItemStart {
+  type: 'getItemStart';
+  contentId: string;
```

Comment:
Is the `contentId` in any way related to a Saved Object id if the content is a saved object? If so, given that we're using the exClient directly, would the Saved Object id be the raw ES id?

Another question related to saved object content types: How are spaces and space-specific objects handled? What about shared saved objects? I guess with shared types we can assume uniqueness.

Reply:
> Is the contentId in any way related to a Saved Object id if the content is a saved object? If so, given that we're using the exClient directly, would the Saved Object id be the raw ES id?

The `id` could be directly SO id and thus the raw ES id, yes. For content management it is just a string passed down to the storage layer to retrieve a content.

> How are spaces and space-specific objects handled? What about shared saved objects? I guess with shared types we can assume uniqueness.

The CM layer is DB agnostic it does not know about SO. What you are describing is a logic handled by the application. If an application ask to fetch id `1234`, CM will forward that to the storage layer and fetch that content on the storage layer. 
For example, we are on space `A` and the application fetches object with id `1234` that _only_ exist in space `B`, that's an error at the application level. It should not fetch that object. Now, the storage layer has access to the `core` http request context so which, from what I understand, provide security and authorization for SO. That context can be leveraged to prevent the above scenario.

Reply:
> directly SO id and thus the raw ES id,

Saved object IDs go through a [`SavedObjectsSerializer` ](https://github.com/elastic/kibana/blob/main/packages/core/saved-objects/core-saved-objects-base-server-internal/src/serialization/serializer.ts#L28)class that transforms the raw es doc id to an id in the context of Kibana.

Reply:
That is internal to SO and CM does not know (or need to know) about it. Here we are talking about the SO id, if it is equal or not to the raw ES id is internal to SO.